### PR TITLE
Token blacklist

### PR DESCRIPTION
**Pull request by wojtus7**
This is pretty big PR that introduce pinned/hidden coins functionality

More about what it does here:
https://linear.app/issue/RAI-183/add-token-blacklist-to-hide-known-shitcoin-airdrops

I tried to implement with reanimated but animated turned out to be more performant, and solve all performance issues I was facing during development

### CODE REVIEW

**Comment by christianbaroni** [comment]
@wojtus7 great work on this

starting a list here of cleanup items:

- [ ] when in edit mode, tapping a coin row should check/uncheck that row instead of opening its expanded state
- [ ]  header buttons and other sections (collectibles, investments) should animate to 0.4 opacity when in edit mode and be disabled (touch events blocked)
- [ ]  fabs should be disabled in edit mode
- [ ]  can this row become a sticky header in edit mode? https://www.dropbox.com/s/1xzk81f2ibq4gdv/Screen%20Shot%202020-03-29%20at%205.43.21%20PM.png

File: src/components/coin-row/CoinCheckButton.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+}) => {
+  const [toggle, setToggle] = useState(false);
+  const [previousPinned, setPreviousPinned] = useState(0);
+
+  useEffect(() => {
+    if (toggle && (recentlyPinnedCount > previousPinned || !isCoinListEdited)) {
+      setPreviousPinned(recentlyPinnedCount);
+      setToggle(false);
+    }
+  }, [isCoinListEdited, recentlyPinnedCount]);
```
Comment (christianbaroni) [inline src/components/coin-row/CoinCheckButton.js]: This a valid warning @wojtus7?
Screen Shot 2020-03-29 at 8 14 16 PM

This is the change it's proposing:
`}, [isCoinListEdited, previousPinned, recentlyPinnedCount, toggle]);`

Comment (wojtus7) [reply src/components/coin-row/CoinCheckButton.js]: Resolved in future changes 🎉
With ```recentlyPinnedCount, toggle``` added it rendered more often than necessary :<

**Comment by wojtus7** [comment]
@christianbaroni I think all changes except 4) are applied.
I looked at OfflineToast but it's implemented using reanimated I couldn't make it work considering fact edit mode is created with react-native's Animated 😔

I suggest to leave it animate together for now so it won't block the whole feature and wait for new turbo-modules powered reanimated version to make it performant, and create some `improvement` ticket for that

**Comment by christianbaroni** [comment]
@wojtus7 just tested again, everything looks great!

re: #4, can we use Animated to hide the fabs?

if not, we can just hide them instantly, but animating them down off the screen would be ideal

**Comment by christianbaroni** [comment]
one more: hidden tokens should be hidden from all asset lists (send, swap, deposit)

**Comment by wojtus7** [comment]
@christianbaroni all done, and ready to test 🎉
I totally misunderstood the #4 thinking all coin tabs should be animated or something 🤦‍♂️

When I realized what you meant I was like: Well I'm an idiot 🤦‍♂️🤦‍♂️🤦‍♂️
Well, fabs animate down off the screen now

**Comment by christianbaroni** [comment]
just tested again after @jinchung's changes, everything looks great

one thing I missed before @wojtus7, you can still swipe left and right to profile/camera screens if you swipe in between the header buttons. is there an easy way to disable swiping there?

**Comment by wojtus7** [comment]
@christianbaroni just blocked out all swipe gestures in the header

**Comment by christianbaroni** [comment]
@wojtus7 mind resolving conflicts with develop? then we'll get this merged!

**Pull request merged**